python/cudf/cudf/core/groupby/groupby.py

```diff
@@ -568,6 +570,31 @@ def agg(self, func):
 
         if self._sort:
             result = result.sort_index()
+        else:
+            if cudf.get_option(
+                "mode.pandas_compatible"
+            ) and not libgroupby._is_all_scan_aggregate(normalized_aggs):
+                # Even with `sort=False`, pandas clearly documents and
+                # ensures that the ordering inside the group by columns
+                # is preserved(i.e., it is not non-deterministic).
+                if not isinstance(result_index, cudf.MultiIndex):
+                    lcol, rcol = _match_join_keys(
+                        self.grouping.keys.drop_duplicates()._column,
+                        result_index._column,
+                        "inner",
```

Comment:
Do we need an inner join, or a left join?

Reply:
Switched to left join, yes left join is sufficient in this case.